node/service/src/parachains_db/upgrade.rs

```diff
@@ -102,3 +125,79 @@ fn migrate_from_version_0_to_1(path: &Path) -> Result<(), Error> {
 
 	Ok(())
 }
+
+// This currently clears columns which had their configs altered between versions.
+// The columns to be changed are constrained by the `allowed_columns` vector.
+fn paritydb_fix_columns(
+	path: &Path,
+	options: parity_db::Options,
+	allowed_columns: Vec<u32>,
+) -> io::Result<()> {
+	// Figure out which columns to delete. This will be determined by inspecting
+	// the metadata file.
+	if let Ok(Some(metadata)) = parity_db::Options::load_metadata(&path) {
+		let columns_to_clear = metadata
+			.columns
+			.into_iter()
+			.enumerate()
+			.filter(|(idx, _)| allowed_columns.contains(&(*idx as u32)))
+			.filter_map(|(idx, opts)| {
+				let changed = opts != options.columns[idx];
+				if changed {
+					gum::debug!(
+						target: LOG_TARGET,
+						"Column {} will be cleared. Old options: {:?}, New options: {:?}",
+						idx,
+						opts,
+						options.columns[idx]
+					);
+					Some(idx)
+				} else {
+					None
+				}
+			})
+			.collect::<Vec<_>>();
+
+		if columns_to_clear.len() > 0 {
+			gum::debug!(
+				target: LOG_TARGET,
+				"Database column changes detected, need to cleanup {} columns.",
+				columns_to_clear.len()
+			);
+		}
+
+		for column in columns_to_clear {
+			gum::debug!(target: LOG_TARGET, "Clearing column {}", column,);
+			parity_db::clear_column(path, column.try_into().expect("Invalid column ID"))
+				.map_err(|e| other_io_error(format!("Error clearing column {:?}", e)))?;
+		}
+
+		// Write the updated column options.
+		options
+			.write_metadata(path, &metadata.salt)
+			.map_err(|e| other_io_error(format!("Error writing metadata {:?}", e)))?;
+	}
+
+	Ok(())
+}
+
+/// Migration from version 0 to version 1.
+/// Cases covered:
+/// - upgrading from v0.9.23 or earlier -> the `dispute coordinator column` was changed
+/// - upgrading from v0.9.24+ -> this is a no op assuming the DB has been manually fixed as per
+/// release notes
+fn paritydb_migrate_from_version_0_to_1(path: &Path) -> Result<(), Error> {
+	let mut options = parity_db::Options::with_columns(&path, super::columns::NUM_COLUMNS as u8);
+	for i in columns::ORDERED_COL {
+		options.columns[*i as usize].btree_index = true;
+	}
```

Comment:
can we extract the option creation into a function to dedup with `open_creating_paritydb` in case it changes later?